src/server/requestHandlers.js

```diff
@@ -1,431 +1,490 @@
+                });
+            }
+        })["finally"](function () {
+            var storyModel = JSON.parse(request.req.body.model);
+            var binaryRenameMap = sjrk.storyTelling.server.buildBinaryRenameMap(storyModel.content, request.req.files.file);
+
+            sjrk.storyTelling.server.saveStoryToDatabase(dataSource, binaryRenameMap, storyModel, request.events.onSuccess, request.events.onError);
+        });
+    } else {
+        request.events.onError.fire({
+            isError: true,
+            message: "Saving is currently disabled."
+        });
+    }
+};
+
+// Persist the story model to couch, with the updated
+// references to where the binaries are saved
+sjrk.storyTelling.server.saveStoryToDatabase = function (dataSource, binaryRenameMap, storyModel, successEvent, failureEvent) {
+    var id = uuidv1();
+
+    dataSource.set({directStoryId: id}, storyModel).then(function (response) {
+        response.binaryRenameMap = binaryRenameMap;
+        successEvent.fire(JSON.stringify(response));
+    }, function (error) {
+        failureEvent.fire({
+            isError: true,
+            message: error.reason || "Unspecified server error saving story to database."
+        });
+    });
+};
+
+// Update any media URLs to refer to the changed file names
+sjrk.storyTelling.server.buildBinaryRenameMap = function (blocks, files) {
+    // key-value pairs of original filename : generated filename
+    // this is used primarily by tests, but may be of use
+    // to client-side components too
+    var binaryRenameMap = {};
+
+    fluid.each(blocks, function (block) {
+        if (block.blockType === "image" || block.blockType === "audio" || block.blockType === "video") {
+            if (block.fileDetails) {
+                // Look for the uploaded file matching this block
+                var mediaFile = fluid.find_if(files, function (file) {
+                    return file.originalname === block.fileDetails.name;
+                });
+
+                // If we find a match, update the media URL. If not, clear it.
+                if (mediaFile) {
+                    sjrk.storyTelling.server.setMediaBlockUrl(block, mediaFile.filename);
+                    binaryRenameMap[mediaFile.originalname] = mediaFile.filename;
+                } else {
+                    sjrk.storyTelling.server.setMediaBlockUrl(block, null);
+                }
+            } else {
+                sjrk.storyTelling.server.setMediaBlockUrl(block, null);
+            }
+        }
+    });
+
+    return binaryRenameMap;
+};
+
+// Rotates an image to be oriented based on its EXIF orientation data, if present
+sjrk.storyTelling.server.rotateImageFromExif = function (file, options) {
+    var togo = fluid.promise();
+
+    try {
+        // ensure the file is present and we have all permissions
+        fse.accessSync(file.path);
+
+        // apply default rotation options if none are provided
+        options = options || { quality: 85 };

```

Comment:
Can you add a code comment to explain why the default rotation options is set to `{ quality: 85 }`? Thanks.

Reply:
It was an arbitrary value I chose which was of decently high quality without going for 100%, which would generate the largest files. Is that a reasonable thing to put in a comment?

The default for quality when I call `jo.rotate` is 100, which I suppose is also okay. In that case, I would simply remove this line altogether.